.github/workflows/release.yml

```diff
@@ -0,0 +1,37 @@
+name: Release
+
+on:
+  release:
+    types:
+      - published
+
+env:
+  DEFAULT_PYTHON: 3.9
+
+jobs:
+  release-pypi:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code from Github
+        uses: actions/checkout@v2.3.4
+      - name: Set up Python ${{ env.DEFAULT_PYTHON }}
+        id: python
+        uses: actions/setup-python@v2.2.2
+        with:
+          python-version: ${{ env.DEFAULT_PYTHON }}
+      - name: Install requirements
+        run: |
+          python -m pip install -U pip twine wheel
+          python -m pip install -U "setuptools>=56.0.0"
+      - name: Build distributions
+        run: |
+          python setup.py sdist bdist_wheel
+      - name: Upload to PyPI
+        if: github.event_name == 'release' && startsWith(github.ref, 'refs/tags')
```

Comment:
Also note this line here

Reply:
The release on github should start with a tag ? For example I pushed ``2.6.0-dev0`` as a tag right now, do I need to release a version, or is there something else ? (This is a nice workflow, it forces us to have a release and a tag)

Reply:
> For example I pushed 2.6.0-dev0 as a tag right now, do I need to release a version

You do. The release just needs to for the same commit you tagged. I usually create the tag together with the release, but it should work either way.

Reply:
I believe it shouldn't matter if it's a pre-release.

Reply:
@Pierre-Sassoulas Might be that you need to add the workflow to the master branch first as it isn't trigged on a PR action.

Reply:
OK I'm going to try that and upgrade the release process.